lib/TPP/ToBlockLayoutAndBack.cpp

```diff
@@ -587,17 +601,38 @@ struct PropagateThroughElementWiseOp
     : public OpRewritePattern<linalg::GenericOp> {
   using OpRewritePattern<linalg::GenericOp>::OpRewritePattern;
 
-  // Check sinking preconditions: a) all loops must
-  // be parallel.
-  LogicalResult checkPreconditions(linalg::GenericOp linalgOp) const {
-    if (linalgOp.getNumLoops() != linalgOp.getNumParallelLoops())
+  LogicalResult isElementwise(linalg::GenericOp linalgOp) const {
+    if (!linalg::isElementwise(linalgOp))
```

Comment:
This looks like a nop function, why not just use `linalg::isElementwise` directly?

Reply:
Done.